sigstore/_internal/rekor/client.py

```diff
@@ -411,14 +406,10 @@ def __del__(self) -> None:
         self.session.close()
 
     @classmethod
-    def production(cls) -> RekorClient:
-        return cls(
-            DEFAULT_REKOR_URL, _DEFAULT_REKOR_ROOT_PUBKEY, CTKeyring.production()
-        )
-
-    @classmethod
-    def staging(cls) -> RekorClient:
-        return cls(STAGING_REKOR_URL, _STAGING_REKOR_ROOT_PUBKEY, CTKeyring.staging())
+    def with_updater(cls, updater: TrustUpdater) -> RekorClient:
+        rekor_key = updater.get_rekor_key()
+        ctfe_keys = updater.get_ctfe_keys()
+        return cls(DEFAULT_REKOR_URL, rekor_key, CTKeyring(ctfe_keys))
```

Comment:
This is a bug: should not use a DEFAULT_REKOR_URL here. Need to reconsider

Reply:
Making sure I understand: the bug here is that this URL will be wrong with `with_updater` is used on `TrustUpdater.staging()`, correct?

Reply:
This is causing `sigstore verify --staging` to fail, so I'll look in to it next.

Reply:
More generally, this suggests that we should really have a top-level configuration object, similar to a configurator, that unifies all of this state (since we should always use like-with-like, so there's no need for flexibility here).

Reply:
3a8f026 has the initial workaround.

Reply:
Another option here is to let individual components manage their TUF state -- is there any risk in having `RekorClient.production()` create its own `TrustUpdater`, other than the performance hit with the current implementation?

Alternatively, maybe a `TrustUpdater` singleton.

Reply:
Thanks for investigating.

Making TrustUpdater aware of the sigstore mechanisms like RekorClient seems wrong.... although there are edge cases like CTKeyring thatare more data types that could indeed be return values for TrustUpdater.

I think the original idea is still salvageable -- I just removed the RekorClient.production()/staging() constructors too greedily. Creating a single TrustUpdater and supplying that "down the call stack" should work (there are only a few places where it's needed so it's not too much work).
 
I will try modifying the RekorClient constructors and will ask you to review if it seems to work.